frontend/src/pages/modelServing/screens/projects/utils.ts

```diff
@@ -176,3 +176,13 @@ export const useCreateInferenceServiceObject = (
 
   return createInferenceServiceState;
 };
+
+export const filterOutConnectionsWithoutBucket = (
+  connections: DataConnection[],
+): DataConnection[] => {
+  if (connections.length === 0) {
+    return [];
+  }
+
+  return connections.filter((obj) => obj.data?.data?.['AWS_S3_BUCKET']?.trim() !== '');
```

Comment:
Probably some unnecessary optional chaining used here, I logged a tech debt to clean up it https://github.com/opendatahub-io/odh-dashboard/issues/1977